bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java

```diff
@@ -0,0 +1,504 @@
+            }
+        } else if (episodeItem != null) {
+            runItem(episodeItem, playCommand);
+        } else {
+            logger.warn("Nothing to display for: {}", terms);
+        }
+    }
+
+    private void runSeriesEpisode(String terms, int season, int episode, @Nullable PlayCommand playCommand)
+            throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        logger.debug("{} series episode mode", playCommand != null ? "Play" : "Browse");
+        var seriesItem = getServerHandler().searchItem(terms, "Series", null);
+        if (seriesItem != null) {
+            logger.debug("Searching series {} episode {}x{}", seriesItem.getName(), season, episode);
+            var episodeItem = getServerHandler().getSeriesEpisodeItem(seriesItem.getId(), season, episode);
+            if (episodeItem != null) {
+                runItem(episodeItem, playCommand);
+            } else {
+                logger.warn("Series {} episode {}x{} not found", seriesItem.getName(), season, episode);
+            }
+        } else {
+            logger.warn("Series not found");
+        }
+    }
+
+    private void runItem(BaseItemDto item, @Nullable PlayCommand playCommand)
+            throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        var itemType = Objects.requireNonNull(item.getType());
+        logger.debug("{} {} '{}'", playCommand == null ? "Browsing" : "Playing", itemType.toLowerCase(),
+                "Episode".equals(itemType) ? item.getSeriesName() + ": " + item.getName() : item.getName());
+        if (playCommand == null) {
+            browseItem(item);
+        } else {
+            playItem(item, playCommand);
+        }
+    }
+
+    private void playItem(BaseItemDto item, PlayCommand playCommand)
+            throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        playItem(item, playCommand, null);
+    }
+
+    private void playItem(BaseItemDto item, PlayCommand playCommand, @Nullable Long startPositionTicks)
+            throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        if (playCommand.equals(PlayCommand.PLAY_NOW)) {
+            stopCurrentPlayback();
+        }
+        getServerHandler().playItem(lastSessionId, playCommand, item.getId().toString(), startPositionTicks);
+    }
+
+    private void browseItem(BaseItemDto item)
+            throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        stopCurrentPlayback();
+        getServerHandler().browseToItem(lastSessionId, Objects.requireNonNull(item.getType()), item.getId().toString(),
+                Objects.requireNonNull(item.getName()));
+    }
+
+    private void stopCurrentPlayback() throws SyncCallback.SyncCallbackError, ApiClientException, InterruptedException {
+        if (lastPlayingState) {
+            sendPlayStateCommand(PlaystateCommand.STOP);
+            Thread.sleep(2000);
```

Comment:
What's the purpose of this sleep?

Reply:
I need to give some time to the client to stop the current media or the play request gets ignored.

Reply:
OK. This implementation will block a thread of the framework from the thread pool that is shared between all bindings. You could schedule a task via the provided scheduler to issue the play request.